packages/plugin-core/src/commands/NoteLookupCommand.ts

```diff
@@ -166,6 +166,10 @@ export class NoteLookupCommand
     return this._provider;
   }
 
+  public set provider(provider: ILookupProviderV3 | undefined) {
```

Comment:
Completely agree. Normally I would go with the typical flow of creating a lookup-derived command (`[[Add a new command that uses lookup|dendron://dendron.docs/pkg.plugin-core.t.lookup.cook#add-a-new-command-that-uses-lookup]]`) and handle it within the new command, but then I will be duplicating the `acceptItem` behavior which I actually don't touch at all. So I went with the route of creating a new provider with the opts I needed, and swapping it out when triggering the already existing lookup command instance.

Reply:
Lookup is currently only modular until the controller / provider level, so a remedy for this would be splitting out all individual core chunks of logic so that they could be ingredients that we can use on a command-to-command basis